source/modulo_components/include/modulo_components/ComponentInterface.hpp

```diff
@@ -601,6 +599,44 @@ inline state_representation::CartesianPose ComponentInterface<NodeT>::lookup_tra
   return result;
 }
 
+template<class NodeT>
+inline void ComponentInterface<NodeT>::publish_predicates() {
+  for (const auto& predicate: this->predicates_) {
+    std_msgs::msg::Bool msg;
+    msg.data = this->get_predicate(predicate.first);
+    if (this->predicate_publishers_.find(predicate.first) == this->predicate_publishers_.end()) {
+      RCLCPP_ERROR_STREAM_THROTTLE(this->get_logger(), *this->get_clock(), 1000,
+                                   "No publisher for predicate " << predicate.first << " found.");
+      return;
+    }
+    predicate_publishers_.at(predicate.first)->publish(msg);
+  }
+}
+
+template<class NodeT>
+inline void ComponentInterface<NodeT>::publish_outputs() {
+  for (const auto& publisher: this->outputs_) {
```

Comment:
For this function and the daemon call-backs, when you need to access both elements of the pair, I would prefer to use structured bindings

```c++
for (const auto& [signal, publisher]: this->outputs_)
```